src/storage/src/source/postgres/snapshot.rs

```diff
@@ -352,16 +404,53 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
 
             let upstream_info = upstream_info.into_iter().map(|t| (t.oid, t)).collect();
 
+            record_table_sizes(
+                id,
+                worker_id,
+                &connection_config,
+                &config.config.connection_context,
+                &snapshot,
+                config
+                    .config
+                    .parameters
+                    .pg_source_snapshot_statement_timeout,
+                metrics,
+                config.config.parameters.pg_snapshot_config,
+                reader_snapshot_table_info
+                    .iter()
+                    .map(|(_, (_, desc, _))| {
+                        (
+                            format!(
+                                "{}.{}",
+                                Ident::new_unchecked(desc.namespace.clone()).to_ast_string(),
+                                Ident::new_unchecked(desc.name.clone()).to_ast_string()
+                            ),
+                            desc.oid.clone(),
+                        )
+                    })
+                    .collect(),
```

Comment:
Will do an in-depth review tomorrow but this is a monster call. I suggest we pass a reference to `config` as a whole. This will collapse 5 parameters (`id`, `worker_id`, `config.config.connection_context`, `config.config.parameters.pg_source_snapshot_statement_timeout`, `config.config.parameters.pg_snapshot_config`) into one. Also, we can calculate the `tables` argument in a separate line before the call. Finally, I would suggest only keeping the option where each worker is dealing with the tables it's assigned. It's doesn't feel right that we distribute the snapshot workload unconditionally but collecting the metrics can happen from a different worker. Each table has a certain amount of work that needs to be done and so the assigned worker should perform it

Reply:
Passing `config` seems like a good plan to me! (that and the `tables` change are implemented and pushed to this pr)

I agree that we may decide, after we see some `count(*)` latencies that just want serial execution, but I am tempted to leave the control here, just in case we truly want to play with these. With the changes that fixed the ci ooms, I believe the complexity is not too bad. That said, do you think that in the case that we should ensure we reuse the existing `Client` when performing this serially? Currently I always open a new connection.

Reply:
>  but I am tempted to leave the control here, just in case we truly want to play with these.

If you have high confidence that we will indeed try out different options and collect statistics to inform our decisions, sounds good. If the confidence is low.. I don't know, I would just do the simple thing. Up to you, I don't have a strong opinion on this. 

> That said, do you think that in the case that we should ensure we reuse the existing Client when performing this serially?

I'd do it if it's easy to write otherwise multiple clients sound fine to me.

Finally, I still see no future where we perform the statistic collection (serial or not) only from the leader worker so I would remove this option. We unconditionally snapshot the whole table from a particular worker which is a much more heavyweight operation. If we ever want to limit snapshot parallelism we will do it for everything (snapshot + statistics) and not piecemeal by changing the exchange rules.

Reply:
Sounds good!

I added 2 commits, the first does:
- reuse the replication client when waiting for the count
- change the default to wait for the count

The second remove thes `collect_count_per_worker`, defaulting to doing it per-worker always.